cdc/sink/table_sink.go

```diff
@@ -16,18 +16,21 @@ package sink
 import (
 	"context"
 	"sort"
+	"sync"
 
 	"github.com/pingcap/errors"
 	"github.com/pingcap/log"
 	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/pkg/container/queue"
 	"github.com/prometheus/client_golang/prometheus"
 	"go.uber.org/zap"
 )
 
 type tableSink struct {
 	tableID     model.TableID
 	backendSink Sink
-	buffer      []*model.RowChangedEvent
+	buffer      *queue.ChunkQueue[*model.RowChangedEvent]
+	bufferMu    sync.Mutex
```

Comment:
Why do we need a lock here? Why doesn't the old one need it?

Reply:
Will it lead to performance regressions? I think using a Mutex is safer here.

Reply:
have removed the mutex